src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -943,6 +943,26 @@ trait CudaOps extends Dsl with StackArrayOps with SizeTOps with CLibs with CudaF
   }
 }
 
+trait CudaLibs extends CudaOps {
+  def cudaEmbedding[T:Numeric:Manifest](implicit __pos: SourceContext) = cudaGlobalFun {
+    (embedding: Rep[Array[T]], indices: Rep[Array[Int]], output: Rep[Array[T]], embed_size: Rep[Int]) => {
+      generate_comment("this is cuda embedding kernel.")
+      generate_comment("arg0: 2D embedding table: <n_embedding x embed_size>")
+      generate_comment("arg1: 1D indices: <indices_size>")
+      generate_comment("arg2: 2D output: <indices_size x embed_size>")
+      generate_comment("arg3: embed_size")
+      generate_comment("invocation assumption: <<<dim3(a,1,1), dim3(indices_size,1,1)>>> where a <= embed_size")
+      generate_comment("each thread block handles one embedding vector")
```

Comment:
good habit to give a list of comments that will show up in the generated kernel and explain the calling conventions

Reply:
will do the same for other kernels.